frontend/src/components/fileUploader.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { ChangeEvent, useState } from 'react';
+import styles from './fileUploader.module.css';
+import UploadIcon from '../icons/upload.svg';
+import UploadInProgressIcon from '../icons/upload-in-progress.svg';
+import CheckCircleIcon from '../icons/check-circle.svg';
+
+interface FileUploaderProps {
+  onFileUpload: (file: File) => Promise<void>;
+  isUploading: boolean;
+  isUploaded: boolean;
```

Comment:
```suggestion
  uploadInProgress: boolean;
  uploadComplete: boolean;
```
I would suggest these names are slightly nicer to read

Reply:
Done :)